src/main/java/com/eternalcode/parcellockers/gui/ParcelItemStorageGUI.java

```diff
@@ -0,0 +1,76 @@
+package com.eternalcode.parcellockers.gui;
+
+import com.eternalcode.parcellockers.configuration.implementation.PluginConfiguration;
+import com.eternalcode.parcellockers.parcel.ParcelSize;
+import dev.triumphteam.gui.guis.Gui;
+import dev.triumphteam.gui.guis.GuiItem;
+import dev.triumphteam.gui.guis.StorageGui;
+import net.kyori.adventure.text.minimessage.MiniMessage;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+
+public class ParcelItemStorageGUI {
+
+    private final PluginConfiguration config;
+    private final MiniMessage miniMessage;
+
+    private StorageGui gui;
+    private boolean confirmed;
+
+    public ParcelItemStorageGUI(PluginConfiguration config, MiniMessage miniMessage) {
+        this.config = config;
+        this.miniMessage = miniMessage;
+    }
+
+    void show(Player player, ParcelSize size) {
+        GuiItem backgroundItem = this.config.guiSettings.mainGuiBackgroundItem.toGuiItem(this.miniMessage);
+        GuiItem confirmItem = this.config.guiSettings.confirmItemsItem.toGuiItem(this.miniMessage, event -> {
+            this.confirmed = true;
+            event.getView().close();
+            new ParcelSendingGUI(this.config, this.miniMessage).show(player);
+        });
+        GuiItem cancelItem = this.config.guiSettings.cancelItemsItem.toGuiItem(this.miniMessage, event -> {
+            this.confirmed = false;
+            event.getView().close();
+            new ParcelSendingGUI(this.config, this.miniMessage).show(player);
```

Comment:
After closing the player view, you open a new GUI.
```suggestion
        GuiItem confirmItem = this.config.guiSettings.confirmItemsItem.toGuiItem(this.miniMessage, event -> {
            this.confirmed = true;
            new ParcelSendingGUI(this.config, this.miniMessage).show(player);
        });
        GuiItem cancelItem = this.config.guiSettings.cancelItemsItem.toGuiItem(this.miniMessage, event -> {
            this.confirmed = false;
            new ParcelSendingGUI(this.config, this.miniMessage).show(player);
```

Reply:
Good point